pkg/controlplane/apiserver/aggregator.go

```diff
@@ -67,3 +279,55 @@ func DefaultGenericAPIServicePriorities() map[schema.GroupVersion]APIServicePrio
 		// Version can be set to 9 (to have space around) for a new group.
 	}
 }
+
+func apiServicesToRegister(delegateAPIServer genericapiserver.DelegationTarget, registration autoregister.AutoAPIServiceRegistration, apiVersionPriorities map[schema.GroupVersion]APIServicePriority) []*v1.APIService {
+	apiServices := []*v1.APIService{}
+
+	for _, curr := range delegateAPIServer.ListedPaths() {
+		if curr == "/api/v1" {
+			prio := apiVersionPriorities[schema.GroupVersion{Group: "", Version: "v1"}]
+			apiService := &v1.APIService{
+				ObjectMeta: metav1.ObjectMeta{Name: "v1."},
+				Spec: v1.APIServiceSpec{
+					Version:              "v1",
+					GroupPriorityMinimum: prio.Group,
+					VersionPriority:      prio.Version,
+				},
+			}
+			registration.AddAPIServiceToSyncOnStart(apiService)
+			apiServices = append(apiServices, apiService)
+			continue
+		}
+
+		if !strings.HasPrefix(curr, "/apis/") {
+			continue
+		}
+		// this comes back in a list that looks like /apis/rbac.authorization.k8s.io/v1alpha1
+		tokens := strings.Split(curr, "/")
+		if len(tokens) != 4 {
+			continue
+		}
+
+		gv := schema.GroupVersion{Group: tokens[2], Version: tokens[3]}
+		prio, ok := apiVersionPriorities[gv]
+		if !ok {
+			// if we aren't found, then we shouldn't register ourselves because it could result in a CRD group version
+			// being permanently stuck in the APIServices list.
+			klog.Infof("Skipping APIService creation for %v", gv)
+			return nil
+		}
+		apiService := &v1.APIService{
+			ObjectMeta: metav1.ObjectMeta{Name: gv.Version + "." + gv.Group},
+			Spec: v1.APIServiceSpec{
+				Group:                gv.Group,
+				Version:              gv.Version,
+				GroupPriorityMinimum: prio.Group,
+				VersionPriority:      prio.Version,
+			},
+		}
+		registration.AddAPIServiceToSyncOnStart(apiService)
+		apiServices = append(apiServices, apiService)
+	}
+
+	return apiServices
+}
```

Comment:
-func apiServicesToRegister(delegateAPIServer genericapiserver.DelegationTarget, registration autoregister.AutoAPIServiceRegistration) []*v1.APIService {
+func apiServicesToRegister(delegateAPIServer genericapiserver.DelegationTarget, registration autoregister.AutoAPIServiceRegistration, apiVersionPriorities map[schema.GroupVersion]APIServicePriority) []*v1.APIService {
 	apiServices := []*v1.APIService{}

 	for _, curr := range delegateAPIServer.ListedPaths() {
 		if curr == "/api/v1" {
-			apiService := makeAPIService(schema.GroupVersion{Group: "", Version: "v1"})
+			prio := apiVersionPriorities[schema.GroupVersion{Group: "", Version: "v1"}]
+			apiService := &v1.APIService{
+				ObjectMeta: metav1.ObjectMeta{Name: "v1."},
+				Spec: v1.APIServiceSpec{
+					Version:              "v1",
+					GroupPriorityMinimum: prio.Group,
+					VersionPriority:      prio.Version,
+				},
+			}
 			registration.AddAPIServiceToSyncOnStart(apiService)
 			apiServices = append(apiServices, apiService)
 			continue
@@ -18,9 +26,22 @@
 			continue
 		}

-		apiService := makeAPIService(schema.GroupVersion{Group: tokens[2], Version: tokens[3]})
-		if apiService == nil {
-			continue
+		gv := schema.GroupVersion{Group: tokens[2], Version: tokens[3]}
+		prio, ok := apiVersionPriorities[gv]
+		if !ok {
+			// if we aren't found, then we shouldn't register ourselves because it could result in a CRD group version
+			// being permanently stuck in the APIServices list.
+			klog.Infof("Skipping APIService creation for %v", gv)
+			return nil
+		}
+		apiService := &v1.APIService{
+			ObjectMeta: metav1.ObjectMeta{Name: gv.Version + "." + gv.Group},
+			Spec: v1.APIServiceSpec{
+				Group:                gv.Group,
+				Version:              gv.Version,
+				GroupPriorityMinimum: prio.Group,
+				VersionPriority:      prio.Version,
+			},
 		}
 		registration.AddAPIServiceToSyncOnStart(apiService)
 		apiServices = append(apiServices, apiService)